src/main/java/com/exasol/adapter/dynamodb/queryrunner/DynamodbGetItemQueryPlanFactory.java

```diff
@@ -0,0 +1,114 @@
+ */
+public class DynamodbGetItemQueryPlanFactory {
+
+    /**
+     * Builds a {@link DynamodbGetItemQueryPlan} is possible for the given query.
+     * 
+     * @param documentQuery query to build the plan for
+     * @param tableMetadata DynamoDB table metadata used for checking the primary key
+     * @return the generated plan
+     * @throws PlanDoesNotFitException if this query can't be executed using a {@link DynamodbGetItemQueryPlan}
+     */
+    public DynamodbGetItemQueryPlan buildGetItemPlanIfPossible(
+            final RemoteTableQuery<DynamodbNodeVisitor> documentQuery, final DynamodbTableMetadata tableMetadata)
+            throws PlanDoesNotFitException {
+        final Visitor visitor = new Visitor(tableMetadata);
+        try {
+            documentQuery.getSelection().accept(visitor);
+            final Map<String, AttributeValue> key = visitor.getPrimaryKey();
+            checkIfKeyIsComplete(key, tableMetadata);
+            return new DynamodbGetItemQueryPlan(documentQuery.getFromTable().getRemoteName(), key);
+        } catch (final PlanDoesNotFitExceptionWrapper wrapper) {
+            throw wrapper.getWrappedException();
+        }
+    }
+
+    private void checkIfKeyIsComplete(final Map<String, AttributeValue> key, final DynamodbTableMetadata tableMetadata)
+            throws PlanDoesNotFitException {
+        if (key.size() < 1) {
+            throw new PlanDoesNotFitException(
+                    "Not a GetItem request as the partition key was not specified in the where clause.");
+        }
+        if (tableMetadata.getPrimaryKey().hasSortKey() && key.size() != 2) {
+            throw new PlanDoesNotFitException(
+                    "Not a GetItem request as the sort key was not specified in the where clause.");
+        }
+    }
+
+    private static class Visitor implements QueryPredicateVisitor<DynamodbNodeVisitor> {
+        private final DynamodbTableMetadata tableMetadata;
+        private final Map<String, AttributeValue> primaryKey = new HashMap<>();
+
+        private Visitor(final DynamodbTableMetadata tableMetadata) {
+            this.tableMetadata = tableMetadata;
+        }
+
+        @Override
+        public void visit(
+                final ColumnLiteralComparisonPredicate<DynamodbNodeVisitor> columnLiteralComparisonPredicate) {
+            final DocumentPathExpression columnPath = columnLiteralComparisonPredicate.getColumn()
+                    .getPathToSourceProperty();
+            if (columnPath.size() != 1) {
+                return; // This is not an key attribute as it
+            }
+            final AttributeValue attributeValue = new DynamodbNodeToAttributeValueConverter()
+                    .convertToAttributeValue(columnLiteralComparisonPredicate.getLiteral());
+            tryToAddKey(this.tableMetadata.getPrimaryKey().getPartitionKey(), columnPath, attributeValue);
+            tryToAddKey(this.tableMetadata.getPrimaryKey().getSortKey(), columnPath, attributeValue);
```

Comment:
Because the columnPath and attributeValue are same for partition and sort keys, you can remove the sort key. Then, this will not add addition sort operation to the returned result?

Reply:
No, DynamoDB sort keys do not mean the result is sorted. A DynamoDB primary key can consist of a partition key and a sort key. This are just two attributes. The partition key is a HASH key and distributes the items over the DynamoDB nodes. The sort key is a sorted index on a node. Thats where the name comes from. Here I try to add the same key condition as a sort key or a partition key condition. The methods only add the key if the given attribute corresponds with the keys attribute.

Reply:
:+1: